### PR TITLE
Desktop: BT system tray applet, global labwc options, desktop refactoring

### PR DESCRIPTION
**Pull request by kajusnau**
## Description of changes

1. **Added global labwc options for GTK+ (font, theme, icon theme, color scheme)**  
   - Added configurable options for GTK+ theme, font, icon theme, and color scheme in `labwc.nix`.
     As of this PR, the default values are:
     ```
     theme = "Adwaita";
     iconTheme = "Papirus";
     colorScheme = "prefer-dark";
     fontName = "Cantarell";
     fontSize = "11";
     ```
   - Desktop components, such as Mako, Ewwbar, have been adjusted to use global labwc font settings.
     Most notably, this means that both Ewwbar and Mako now use **Cantarell** as the default font.
     This can be later adjusted to **Inter**, if that is to be the default Ghaf system font.

2. **Changed the default icon theme to Papirus**  
   - Switched the default icon theme to **Papirus**.
   - **Papirus** icon theme is already used by nwg-drawer (Ghaf app launcher), therefore this change should make Ghaf look more unified.
   - Previously used **Adwaita** icon theme did not contain icons for applets (e.g. nm-applet, blueman-applet), and was generally not applied.
   - The default GTK+ color scheme has also been changed to prefer dark variants. This change affects all GTK apps.
   - icon-pack package has been removed in favor of using global icon theme icons by default.

3. **Enabled icon theme support via gsettings**  
   - Enabled icon theme integration via `gsettings`.
   - This seems to be the recommended, if not only, way to consistently apply GTK theme settings system-wide (https://github.com/swaywm/sway/wiki/GTK-3-settings-on-Wayland)
   - More info about the specific gsettings schema key values can be found [here](https://github.com/GNOME/gsettings-desktop-schemas/blob/master/schemas/org.gnome.desktop.interface.gschema.xml.in)

4. **Set default values for XDG base directories in guivm**  
   - Configured previously unavailable default XDG base directory values in `guivm` to align with standard directory structures.
   - https://wiki.archlinux.org/title/XDG_Base_Directory

5. **Added Bluetooth applet to the system tray**  
   - Introduced a Bluetooth applet to the system tray for easy access to Bluetooth settings and device management.
   
6. **Adjusted behavior of Bluetooth notifications**  
   - Bluetooth notification will now be displayed via Mako - meaning all bluetooth notifications will appear in the top left corner along with any other system notifications.

6. **Adjusted behavior of nm-launcher and bt-launcher**
  - nm-applet and blueman-applet will no longer interfere with each other.
  This behavior was previously observed due to both attempting to use a control socket with the same name.

7. **Adjusted Mako notification configuration**  
   - Mako notifications will now appear with their appropriate **Papirus** icon theme icons.
   - Notifications now have a 5 second default timeout (previously 10). 

8. **Moved Bluetooth and nm applet services configuration under labwc config**  
   - Centralized configuration for Bluetooth and Network Manager applets within the `labwc` configuration.  
   - This change aligns with the general approach of declaring desktop services under `labwc` config.

9. **Refactored ghaf launcher implementation**  
   - Removed `ghaf-launcher.nix` - nwg-drawer daemon is now started directly by the ghaf-launcher service.
   - This change should result in a small improvement in overall responsiveness of the Ghaf app launcher.

## Known issues:
- All GTK apps have been changed to prefer dark theme variant, following Ghaf's general UI design.
- Some applications, such as *ghaf audio control*, do not display correctly with the new `prefer-dark` setting for GTK+ apps.
  For Ghaf's own apps, future adjustments may be needed for better compatibility with dark theme preferences.

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [x] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status
- [x] Change requires full re-installation
- [x] Change can be updated with `nixos-rebuild ... switch` - may or may not work reliably, see comments

## Instructions for Testing

- Please test the overall UI look and feel with as many apps as possible, ensuring consistency for dark theme variants.
- Blueman applet should be tested as thoroughly as possible.
- There's been a major overhaul in how app icons are fetched.
  Testing should include checking app launcher icons to verify all icons are present as before and that the icons themselves have not changed.

- [x] List all targets that this applies to:
  Lenovo X1, Orin, VM
- [x] Is this a new feature
  - [x] List the test steps to verify:
  The main new feature to test is blueman applet in the system tray.
  Please test the following:
  - Overall functionality of all submenus options
  - Bluetooth settings (should open from both the applet and Ghaf app launcher)
  - Bluetooth connectivity
- [ ] If it is an improvement how does it impact existing functionality?

### CODE REVIEW

File: modules/desktop/graphics/labwc.nix [hunk 1]
```diff
@@ -95,7 +134,7 @@ in
       [
         pkgs.labwc
         pkgs.ghaf-theme
-        pkgs.adwaita-icon-theme
+        pkgs.papirus-icon-theme
```
[the remark this answers is not in the document]
Comment (kajusnau) [reply modules/desktop/graphics/labwc.nix:137]: Changed, multiple files affected.
Default behavior for application launchers has been adjusted, please check :+1:

**Comment by leivos-unikie** [comment]
I did some preliminary testing with this.

Noticed some strange effect when opening the app menu for the first time, all icons were black, then after opening it again the icons appeared. There seems to be also some problems in closing the app menu (by clicking the launcher icon again). 

In power menu Reboot and Shutdown does not work.

In ci-test-automation I had hard coded some paths to gui icons. Maybe it would be better to search the needed icons files directly to be more independent on the icon theme. It seems that I can find same icon names under breeze then Adwaita had.

```
[ghaf@gui-vm:/run/current-system/sw/share/icons]$ ls
breeze  breeze-dark  hicolor  Papirus  Papirus-Dark  Papirus-Light
```

**Comment by kajusnau** [comment]
> I did some preliminary testing with this.
> 
> Noticed some strange effect when opening the app menu for the first time, all icons were black, then after opening it again the icons appeared. There seems to be also some problems in closing the app menu (by clicking the launcher icon again).
> 
> In power menu Reboot and Shutdown does not work.
> 
> In ci-test-automation I had hard coded some paths to gui icons. Maybe it would be better to search the needed icons files directly to be more independent on the icon theme. It seems that I can find same icon names under breeze then Adwaita had.
> 
> ```
> [ghaf@gui-vm:/run/current-system/sw/share/icons]$ ls
> breeze  breeze-dark  hicolor  Papirus  Papirus-Dark  Papirus-Light
> ```

I cannot reproduce any of the mentioned issues.
That being said, I have run into a similar problem where Reboot and Shutdown didn't work, but only after the Ghaf laptop has been in suspended/sleep state over a long period of time without ever shutting down. In this case, `journalctl | grep eww` might produce some useful logs where some GIVC errors can be observed.
In any case, this power behavior issue is not caused by this PR, but I can double check if there's logs provided.

As for the black icons, this may have been caused by the system attempting to index the new icon theme for the first time, though I am not sure.

I recommend a proper system reboot before proceeding with testing this PR.

**Comment by leivos-unikie** [comment]
> I recommend a proper system reboot before proceeding with testing this PR.

I recall now that power/reboot not working is common symptom after nixos-rebuild switch if the device is not rebooted. I think I clicked the reboot and then just assumed it had rebooted (although it didn't since the button didn't work). Reboot and shutdown works fine now after the machine was shutdown in between. 

Still I have some problems in closing the app menu by clicking the menu launcher icon in the corner. Sometimes I have to click it several times before managing to close. The icon background changes slightly when clicked but the menu does not close, or sometimes the menu closes but then flashes immediately back. I think it's not a hardware issue since the same happens with laptop left button, touchpad and external USB mouse.

**Comment by leivos-unikie** [comment]
> For test automation, the icons should be searched under `$XDG_DATA_DIRS` env var available in gui-vm, something like this can be used:
> 
> ```
> $ find $(echo $XDG_DATA_DIRS | tr ':' ' ') -type d -name "icons" 2>/dev/null
> /run/current-system/sw/share/icons
> $ ls /run/current-system/sw/share/icons
> breeze  breeze-dark  hicolor  Papirus  Papirus-Dark  Papirus-Light
> ```
> 
> Then, to find a specific icon, the currently active icon theme should be searched, `/run/current-system/sw/share/icons/[theme]/` If an icon is not found under the theme, `hicolor` should be searched instead, as per [FreeDesktop specification](https://specifications.freedesktop.org/icon-theme-spec/latest/#directory_layout).

Thanks for guidance with the icons. Is it possible to extract name of the currently active icon theme from ghaf in run-time?

**Comment by leivos-unikie** [comment]
> Still I have some problems in closing the app menu by clicking the menu launcher icon in the corner. Sometimes I have to click it several times before managing to close. The icon background changes slightly when clicked but the menu does not close, or sometimes the menu closes but then flashes immediately back. I think it's not a hardware issue since the same happens with laptop left button, touchpad and external USB mouse.

I have the same problem with app menu launcher also when running ghaf mainline. So after all it is probably something in my HW (if nobody else having similar problems).

**Comment by kajusnau** [comment]
> Thanks for guidance with the icons. Is it possible to extract name of the currently active icon theme from ghaf in run-time?

Unfortunately there is no easy way at the moment. We can address this in a future patch if needed by test automation.

**Comment by kajusnau** [comment]
> I have the same problem with app menu launcher also when running ghaf mainline. So after all it is probably something in my HW (if nobody else having similar problems).

I still cannot reproduce this. It's good to note here that the app launcher toggling behavior works by sending a USR1 signal to a running nwg-drawer process. If Ghaf has trouble sending these signals, problems can occur.

**Comment by leivos-unikie** [comment]
> > Thanks for guidance with the icons. Is it possible to extract name of the currently active icon theme from ghaf in run-time?
> 
> Unfortunately there is no easy way at the moment. We can address this in a future patch if needed by test automation.

I think updating the theme name manually to ci-test-automation is not a big deal if we manage to automate the rest of icon searching. Theme change does not happen every week, I hope.

**Comment by kajusnau** [comment]
Fixed conflicts, rebased.

**Comment by leivos-unikie** [comment]
Tested on Lenovo-X1 (fresh image on USB SSD)

ci-test-automation passed, performance ok

As mentioned in the Known issues Ghaf audio control does not currently show up well with the dark theme. It's difficult to discern white stream labels on the light gray background. 

Dark theme does not apply to:
Calculator
Ghaf control panel? (background has been dark already)
Element
Falcon AI
browsers or apps nested in browser window
Zathura
Sticky Notes
Text Editor
VPN tool
Video Editor? (background has been dark already)
xarchiver

Most of the icons were missing in app menu when first time opened it. This happens only after first boot. I saw the same bug after the first boot also previously when testing this PR. Also Ilkka reproduced this issue.

**Comment by leivos-unikie** [comment]
app_menu

**Comment by leivos-unikie** [comment]
audio_control

**Comment by kajusnau** [comment]
> Dark theme does not apply to: Calculator Ghaf control panel? (background has been dark already) Element Falcon AI browsers or apps nested in browser window Zathura Sticky Notes Text Editor VPN tool Video Editor? (background has been dark already) xarchiver

I suspect the listed apps are either not made with GTK or don't support dynamic themes. In both cases it would be out of our hands.

> Most of the icons were missing in app menu when first time opened it. This happens only after first boot. I saw the same bug after the first boot also previously when testing this PR. Also Ilkka reproduced this issue.

It seems that this happens only once (closing an opening the launcher makes the icons appear), so it's not a major usability problem.

**Comment by leivos-unikie** [comment]
App icons have disappeared also from vm-debug target.

**Comment by kajusnau** [comment]
> Tested on Orin AGX (native build, rebuild switch to remote target)
> 
> * ci-test-automation passed and performance ok
> * App icons are missing totally, only app labels are shown. Icons do not appear even when reopening the app menu or after reboot.
> * Apps can be launched from the labels in app menu

Not sure what is causing the icons not to show up on Orin and VM targets.
In my own testing on VM target, I additionally noticed that nwg-drawer (Ghaf launcher) seems to have a blue tint background instead of the intended black.

**Comment by kajusnau** [comment]
> > Tested on Orin AGX (native build, rebuild switch to remote target)
> > 
> > * ci-test-automation passed and performance ok
> > * App icons are missing totally, only app labels are shown. Icons do not appear even when reopening the app menu or after reboot.
> > * Apps can be launched from the labels in app menu
> 

> Not sure what is causing the icons not to show up on Orin and VM targets. In my own testing on VM target, I additionally noticed that nwg-drawer (Ghaf launcher) seems to have a blue tint background instead of the intended black.

It seems that this was caused by guivm not being used in VM target (and probably Orin), therefore I've moved the XDG session var definitions to labwc.nix. All icons (along with global GTK settings) should now be visible as long as labwc is used by the target in question.

**Comment by leivos-unikie** [comment]
I checked vm-debug first icons are now visible in the menu. The app launcher had at first blueish background but then after testing some time (opening and closing app menu multiple times) I noticed that the background had turned to black.

**Comment by leivos-unikie** [comment]
@barnabakos did testing on Orin AGX, both native and cross-compiled. These are his greetings:

"
PR #888 looks good on AGX cross, but have issues on AGX native:
- app launcher has a blue background, does not turn black
- windows key does not open the app launcher (but it can close it if it is open)
- at least the app icons are visible
"

**Pull request merged**